src/function/aggregate/distributive/sum.cpp

```diff
@@ -129,7 +129,7 @@ unique_ptr<BaseStatistics> SumPropagateStats(ClientContext &context, BoundAggreg
 	return nullptr;
 }
 
-AggregateFunction GetSumAggregate(PhysicalType type) {
+AggregateFunction SumFun::GetSumAggregate(PhysicalType type) {
```

Comment:
I have been wanting to move to a lazy function registration for a while, which would require actually removing the RegisterFunction and replacing it with a method such as this. The idea is similar to what we have in [default_views.cpp](https://github.com/duckdb/duckdb/blob/6c7c9805fdf1604039ebed47d233ea55cabb4b2c/src/catalog/default/default_views.cpp), except we would have a function name and a pointer to a function (e.g. something like {"sum", SumFun::GetSumAggregate }). Then when a lookup is performed in the catalog for a function (e.g. "sum"), we can call that function pointer to register the function lazily. 

This should improve start-up time and make it easier to add functions as you would only need to add the function in one place (this array) rather than having to add the register function method and code that calls that method.